### PR TITLE
Filters: Syntactic separation of ordering and filtering

### PR DESCRIPTION
**Pull request by winged**
We want to separate filtering and ordering syntactically, so it doesn't
make sense to allow filtering in the filter[] list in the first place.

### CODE REVIEW

**Comment by winged** [comment]
Got a good bit ahead today. This is now possible:

```graphql
query new_ordering {
  allDocuments(
    filter: [
    	{metaHasKey: "camac-instance-id"}
    ],
    order: [
      {meta: "foobar", direction: ASC},
      {answerValue: "nutzungsart"}
    ]
  ) {
      ...
  }
}
```

**Comment by winged** [comment]
This should now be ready for review.

Please have a good look at the new order types. Especially:
* Are the filters everywhere you expect them?
* Do the "attribute" filters cover all fields you potentially want to order by?
* Are all existing "orderBy" use cases covered by the new ones?

**Comment by fkm** [comment]
Probable oversights:
* `order` seems to be missing from `allWorkItems`.
* There's an `oderBy` on the `WorkItemFilterSetType`.
* There's an `oderBy` on the `CaseFilterSetType`.
* There's an `orderByQuestionAnswerValue` on the `CaseFilterSetType`.
* There's an `oderBy` on the `QuestionFilterSetType`.

Questions:
* What does `documentForm` do on the `CaseFilterSetType`?
* Will cases be found recursively for `case`, `caseDocumentHasAnswer`, and `caseMetaValue` on  the`WorkItemFilterSetType`?

File: caluma/core/ordering.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    field_class = forms.CharField
+
+    def get_ordering_value(
+        self, qs: QuerySet, value: Any
+    ) -> Tuple[QuerySet, OrderingFieldType]:
+
+        return qs, CombinedExpression(F(self.field_name), "->", Value(value))
+
+
+class AnswerValueOrdering(CalumaOrdering):
```
Comment (open-dynaMIX) [inline caluma/core/ordering.py]: This should be in the `form` app, as `core` should never depend on `form`.

**Comment by winged** [comment]
> Fair enough @ recursive case search! smiley
> 
> Concerning the legacy `orderBy`. Aren't those directly on `allWorkItems` and `allCases`?

They're almost everywhere

> 
> If we had a version with the `filter: { orderBy: "..." }` everything is good +1

No, `foo(filter: [{orderBy..}])` should never exist. If you want the legacy orderBy, go with `foo(orderBy:...)`.

I've explicitly removed them from the `filter:[]` type, as I want this to only do the filtering, not ordering

File: caluma/core/ordering.py (hunk 1 above)
Comment (winged) [reply caluma/core/ordering.py]: Right you are! Fixed

**Pull request merged**